src/java.desktop/share/classes/javax/swing/plaf/metal/MetalLookAndFeel.java

```diff
@@ -252,6 +252,11 @@ public boolean getSupportsWindowDecorations() {
      * used. For example, metal does not have a class named {@code
      * "MetalColorChooserUI"}, as such, {@code
      * javax.swing.plaf.basic.BasicColorChooserUI} is used.
+     * It is to be noted that some MetalXXXUI classes can be used only by
```

Comment:
Seems to me that this needs to be introduced in the previous paragraph otherwise you are revising what is said there in an awkward way.

So we can add to "{@code
     * MetalLookAndFeel} registers an entry for each of the classes in
     * the package {@code javax.swing.plaf.metal} that are named
     * MetalXXXUI"
   just this one extra clause :
", which are used by the current {@code MetalTheme}"

Technically that is ALL that is needed.

But at line 255 we then can (since 252- 254 already set precedent) say
"And {@code MetalMenuBarUI} is an  example of a UI class that is used only by one provided theme : {@code OceanTheme}, and therefore will not be registered by the other provided themes"